tests/test_dataset.rs

```diff
@@ -171,6 +172,88 @@ where
     Ok(())
 }
 
+fn test_byte_read_seek_impl(ds: &hdf5::Dataset, arr: &ArrayD<u8>, ndim: usize) -> hdf5::Result<()> {
+    let mut rng = SmallRng::seed_from_u64(42);
+    ds.write(arr)?;
+
+    // Read whole
+    let ds = ds.clone();
+    let mut reader = match ds.into_byte_reader() {
```

Comment:
This currently fails with
```
error[E0507]: cannot move out of dereference of `_IMPL_H5TYPE_FOR_Enum::_h5::Dataset`
   --> tests/test_dataset.rs:181:28
    |
181 |     let mut reader = match ds.into_byte_reader() {
    |                            ^^^^^^^^^^^^^^^^^^^^^ move occurs because value has type `Container`, which does not implement the `Copy` trait
```
Not quite sure what I am doing wrong.

Reply:
`Container` is hidden behind the deref trait which is a reference. Either do a double deref before clone (`let ds = (**ds).clone()`) or change signature from `into_byte_reader(self)` to `as_byte_reader(&self)` (which I prefer in this case).

Reply:
So then I need to do `(**ds).clone()` in `as_byte_reader()`?

Reply:
It is enough to do a regular clone in `as_byte_reader` as the deref has triggered at the start of the function.